### PR TITLE
7136 infiniteloopbug taketwo

### PR DESCRIPTION
**Pull request by lzach83**
## Link to Issue
Closes: #7136 

## Description of Changes
- Adds check in query to not exceed the limit (20) returns undefined otherwise
- Also added back the hook for resizing, that caused a regression and was not the fix

## "How We Fixed It"

- Added a check in the `getNextPageParam` to return undefined. Before it was just rapidly firing if the end was found. Which is not intended.

## Test Plan
- Go to discussions
- Open dev tools network tab
- On first arrival check to ensure the call isn't rapidly firing
- Do some resizing and ensure there's no regressions.

## Other
[First attempt](https://github.com/hicommonwealth/commonwealth/pull/7138)

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx [hunk 1]
```diff
@@ -117,6 +124,20 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
...
+    if (resizing) {
+      timer = setTimeout(() => {
+        setResizing(false);
+      }, 200); // Adjust delay as needed
+    }
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [resizing]);
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:139]: so this supposed to be a fix? do we know why this helps and what is actually a root cause? This looks sooo hacky and tbh I do not understand it

**Comment by burtonator** [comment]
LGTM but my only feedback is to use a standard debouncer with the resize hook.

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx (hunk 1 above)
Comment (lzach83) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:139]: This is not the fix to the ticket. This was suppose to be here all along to control the `resizing` and `setResizing` 

Before the values were not resetting and they should've been. This was taken from `Sublayout.tsx` and how it's controlled on that level as well.

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx [hunk 2]
```diff
@@ -88,6 +94,7 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
       },
       {
         getNextPageParam: (lastPage) => {
+          if (lastPage.threads.length < 20) return undefined;
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:97]:
```diff
- lastPage.threads.length < 20
+ lastPage.threads?.length < 20
```

**Review by mzparacha — APPROVED** [review]

**Review by rbennettcw — APPROVED** [review]
Works.

Note for those trying to repro: you must reach the last page, so it helps to test on a community with only a few threads.

**Pull request merged**